test/integration/tracked_watermark_buffer.cc

```diff
@@ -0,0 +1,91 @@
+#include "test/integration/tracked_watermark_buffer.h"
```

Comment:
Might be worth it to eventually have some tests for this test code, but up to you.

Reply:
Added.  A followup change adds a few more utility methods to this buffer factory, so having a tested baseline is useful.